src/backend/mpi/swdsm.cpp

```diff
@@ -1199,14 +1199,7 @@ void argo_initialize(unsigned long long size){
 								 MPI_INFO_NULL, MPI_COMM_WORLD, &sharerWindow);
 	MPI_Win_create(lockbuffer, pagesize, 1, MPI_INFO_NULL, MPI_COMM_WORLD, &lockWindow);
 
-	memset(pagecopy, 0, cachesize*pagesize);
-	memset(touchedcache, 0, cachesize);
-	memset(globalData, 0, size_of_chunk*sizeof(argo_byte));
-	memset(cacheData, 0, cachesize*pagesize);
-	memset(lockbuffer, 0, pagesize);
-	memset(globalSharers, 0, gwritersize);
-	memset(cacheControl, 0, cachesize*sizeof(control_data));
-
+	//Sets cache control structures to initial state.
```

Comment:
My understanding is that this part (removing initialization) has nothing to do with the remaining patch.
I am not sure about the performance impacts of this part yet, but maybe we should split this off into a separate pull-request?